fhir-persistence-jdbc/src/main/java/com/ibm/fhir/persistence/jdbc/domain/SearchWholeSystemQuery.java

```diff
@@ -0,0 +1,83 @@
+/*
+ * (C) Copyright IBM Corp. 2021
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package com.ibm.fhir.persistence.jdbc.domain;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import com.ibm.fhir.model.resource.Resource;
+import com.ibm.fhir.persistence.exception.FHIRPersistenceException;
+
+/**
+ * A domain model of the query used to retrieve the count or data
+ * for a particular whole-system search.
+ */
+public class SearchWholeSystemQuery extends SearchQuery {
+    
+    // Set of domain models for resource types used in this whole-system search
+    List<SearchQuery> domainModels;
+
+    // Flag indicating if a count query or a data query
+    boolean isCountQuery;
+
+    // Is pagination required?
+    boolean addPagination = true;
+
+    // Sort parameters
+    final List<DomainSortParameter> sortParameters = new ArrayList<>();
+
+    /**
+     * Public constructor
+     * @param domainModels
+     * @param isCountQuery
+     * @param addPagination
+     */
+    public SearchWholeSystemQuery(List<SearchQuery> domainModels, boolean isCountQuery, boolean addPagination) {
```

Comment:
should we prohibit `isCountQuery=true` if `addPagination=true`?

Reply:
Since pagination (and sort as well) is not applicable for count queries I just updated the code to ignore that flag if isCountQuery=true